keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -0,0 +1,309 @@
+    - [Manually Managed Buckets](#manually-managed-buckets)
+    - [Provisioner Secrets](#provisioner-secrets)
+  - [Custom Resource Definitions](#custom-resource-definitions)
+      - [Bucket](#bucket)
+      - [BucketContent](#bucketcontent)
+      - [BucketClass](#bucketclass)
+<!-- /toc -->
+
+# Summary
+
+This proposal introduces Custom Resource Definitions (CRDs), for the purpose of standardizing object storage representations in Kubernetes.  Goals and non-goals set the scope for the proposal by defining higher level objectives.  The vocabulary section defines terminology.  User stories illustrate how these APIs may fulfill cluster user requirements.  Relationships between the APIs are provied to illustrate the interconnnections between object storage APIs, users' workloads, and object store service instances.  Lastly, the documents states the proposed API specs for the Bucket, BucketContent, and BucketClass objects.
+
+## Motivation
+
+File and block are first class citizens within the Kubernetes ecosystem.  Object, though very different under the hood, is a popular means of storing data, especially against very large data sources.   As such, we feel it is in the interest of the community to elevate buckets to a community supported feature.  In doing so, we can provide Kubernetes cluster users and administrators a normalized and familiar means of managing object storage.
+
+While absolute portability cannot be guaranteed because of incompatibilities between providers, workloads reliant on a given protocol (e.g. one of S3, GCS, Azure Blob) may be defined in a single manifest and deployed wherever that protocol is supported.
+
+The object storage API will also provide a path towards community maintained automatio, which will be capable of bucket lifecycle operations.  It is anticipated that the controller provide an API such that pluggable drivers may be written to implement operations specific to the object store provider.
+
+This proposal does _not_ include a standardized *protocol* or abstraction of storage vendor APIs
+
+## Goals
+
++ Define a control plane API in order to standardize and formalize Kubernetes object storage representation
++ As MVP, be accessible to the largest groups of consumers by supporting the major object storage protocols (S3, Google Cloud Storage, Azure Blob) while being extensible for future protocol additions.
++ Present similar workflows for both greenfield and brownfield bucket operations.
+
+## Non-Goals
+
++ Define the _data-plane_ object store interface.
++ Precisely define automation around proposed bucket APIs
++ Manage user identities or roles
+
+##  Vocabulary
+
++  _Brownfield Bucket_ - externally created and represented by a [BucketClass](#bucketclass) and managed by a provisioner.
++ _Bucket_ - A user-namespaced custom resource representing an object store bucket.
++  _BucketClass_ - A cluster-scoped custom resource containing fields defining the provisioner and an immutable parameter set.
```

Comment:
This overloading of either "A description of how to provision a new bucket" or "A description of a single existing bucket" seems confusing. Why is the design this way?